go-controller/pkg/cluster/node.go

```diff
@@ -122,11 +130,54 @@ func (cluster *OvnClusterController) StartClusterNode(name string) error {
 		return nil
 	}
 
-	err = ovn.CreateManagementPort(node.Name, subnet.String(), clusterSubnets)
+	type readyFunc func(string) (bool, error)
+	var readyFuncs []readyFunc
+
+	macAddress, err := CreateManagementPort(node.Name, subnet.String(), clusterSubnets)
+	if err != nil {
+		return err
+	}
+
+	readyFuncs = append(readyFuncs, ManagementPortReady)
+	numFuncs := len(readyFuncs)
+	wg.Add(numFuncs)
+
+	// Set macAddress as annotation
+	err = cluster.Kube.SetAnnotationsOnNode(node, macAddress)
 	if err != nil {
+		logrus.Errorf("Failed to set node %s macAddress annotation: %v", node.Name, macAddress)
 		return err
 	}
 
+	portName := "k8s-" + node.Name
+
+	// Wait for the portMac to be created
+	for _, f := range readyFuncs {
+		go func(rf readyFunc) {
+			defer wg.Done()
+			if err := wait.PollImmediate(500*time.Millisecond, 2*time.Second, func() (bool, error) {
+				ready, errReady := rf(portName)
```

Comment:
Actually would it work to just:

```
	err := wait.PollImmediate(500*time.Millisecond, 2*time.Second, func() (bool, error) {
		return rf(portName)
	})
	messages <- waitResult{Error: err}
```

If the readyFunc returns `true` that will make PollImmediate return `nil` and that'll have the exact same effect as `messages <- waitResult{}`. If the readyFunc returns an error, or the PollImmediate times out, then it'll return an error and that'll get stuffed into `messages`. I think the above and your code is actually equivalent?